File: seminar_4/task_5/patokova.victoria/task_5.py
class Node:
    def __init__(self, data):
        self.data = data
        self.parent = None
        self.left = None
        self.right = None


class Tree:
    def __init__(self):
        self.root = None


def is_same_tree(a, b):
    if a is None and b is None:
        return True
    if a is None or b is None:
        return False
    if a.data != b.data:
        return False
    
    return is_same_tree(a.left, b.left) and is_same_tree(a.right, b.right)

File: seminar_4/task_5/shatrov.igor/Task5.java
public class Task5 {
    public static void main(String[] args) {

    }

    public static boolean isSameTree(MyBinaryTreeNode a, MyBinaryTreeNode b) {
        if (a == null && b == null) {
            return true;
        }

        if (a == null || b == null) {
            return false;
        }

        if (a.val != b.val) {
            return false;
        }

        return (isSameTree(a.left, b.left) && isSameTree(a.right, b.right));
    }
}
